generators/generator-base.js

```diff
@@ -2267,6 +2267,153 @@ templates: ${JSON.stringify(existingTemplates, null, 2)}`;
+    };
+
+    const allBlocks = Object.entries(files).map(async ([key, section]) => {
+      assert(Array.isArray(section), `Section must be an array for ${key}`);
+      return section.map(async (block, idx) => {
+        assert(typeof block === 'object', `Block must be an object for ${key}[${idx}]`);
+        assert(Array.isArray(block.templates), `Block templates must be an array for ${key}[${idx}]`);
+        const { path: blockPathCallback = './', to: blockToCallback, condition: blockConditionCallback } = block;
+        const condition = await normalizeCallback(blockConditionCallback);
+        if (condition !== undefined && !condition) {
+          return undefined;
+        }
+        const blockPath = await normalizeCallback(blockPathCallback);
+        const blockTo = (await normalizeCallback(blockToCallback)) || blockPath;
+        return block.templates.map(async templateObj => {
+          let templateOptions;
+          let sourceFile;
+          let destinationFile;
+          if (typeof templateObj === 'string') {
+            sourceFile = path.join(blockPath, templateObj);
+            destinationFile = this.destinationPath(blockTo, normalizeEjs(templateObj));
+          } else {
+            const { file, renameTo, destinationFile: objDestinationFile } = templateObj;
+            templateOptions = templateObj.options;
+            const normalizedFile = await normalizeCallback(file);
+            sourceFile = path.join(blockPath, normalizedFile);
+            destinationFile = this.destinationPath(
+              blockTo,
+              path.join(await normalizeCallback(objDestinationFile || renameTo, () => normalizeEjs(normalizedFile)))
+            );
+
+            const override = await normalizeCallback(templateObj.override);
+            if (override !== undefined && !override && this.fs.exists(destinationFile)) {
+              this.debug(`skipping file ${destinationFile}`);
+              return undefined;
+            }
+          }
+
+          const extension = path.extname(sourceFile);
+          const appendEjs = !templateObj.noEjs && !['.ejs', '.png', '.jpg', '.gif', '.svg', '.ico'].includes(extension);
+          const ejsFile = appendEjs || extension === '.ejs';
+
+          let sourceFileFrom;
+          if (Array.isArray(rootTemplatesAbsolutePath)) {
+            // Look for existing templates
+            const existingTemplates = rootTemplatesAbsolutePath
+              .map(rootPath => this.templatePath(rootPath, sourceFile))
+              .filter(templateFile => fs.existsSync(appendEjs ? `${templateFile}.ejs` : templateFile));
+
+            if (existingTemplates.length > 1) {
+              const moreThanOneMessage = `Multiples templates were found for file ${sourceFile}, using the first
+templates: ${JSON.stringify(existingTemplates, null, 2)}`;
+              if (existingTemplates.length > 2) {
+                this.warning(`Possible blueprint conflict detected: ${moreThanOneMessage}`);
```

Comment:
what about introducing a logger?

Reply:
If you think so, yeoman already has a very simple logger, IMO we should keep using it.